src/contracts/core/DelegationManager.sol

```diff
@@ -477,6 +490,11 @@ contract DelegationManager is
                 });
             }
 
+            //Reset DepositScalingFactor since deposit shares are equal to withdrawable shares after full withdrawal
+            if (depositSharesToWithdraw[i] == shareManager.stakerDepositShares(staker, strategies[i])) {
+                _depositScalingFactor[staker][strategies[i]].reset();
+            }
```

Comment:
I would rather have `removeDepositShares` return the new number of deposit shares, and compare that against 0

This way feels more messy.